From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Tue, 10 Apr 2018 17:26:34 +0200
Subject: firmware-utils: tplink-safeloader: move CPE/WBS 210/510 version metainfo to the end

Having the metainfo between kernel and rootfs prevents us from resizing
the kernel partition as necessary.

Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>

diff --git a/tools/firmware-utils/src/tplink-safeloader.c b/tools/firmware-utils/src/tplink-safeloader.c
index 11ff2e56e19ec3780f988baf7257810530165b23..2a2329d1f188ea8520b6a4aeef25937e05d48a06 100644
--- a/tools/firmware-utils/src/tplink-safeloader.c
+++ b/tools/firmware-utils/src/tplink-safeloader.c
@@ -139,10 +139,10 @@ static struct device_info boards[] = {
 			{"default-mac", 0x30000, 0x00020},
 			{"product-info", 0x31100, 0x00100},
 			{"signature", 0x32000, 0x00400},
-			{"os-image", 0x40000, 0x170000},
-			{"soft-version", 0x1b0000, 0x00100},
-			{"support-list", 0x1b1000, 0x00400},
-			{"file-system", 0x1c0000, 0x600000},
+			{"os-image", 0x40000, 0x180000},
+			{"file-system", 0x1c0000, 0x5f0000},
+			{"soft-version", 0x7b0000, 0x00100},
+			{"support-list", 0x7b1000, 0x00400},
 			{"user-config", 0x7c0000, 0x10000},
 			{"default-config", 0x7d0000, 0x10000},
 			{"log", 0x7e0000, 0x10000},
@@ -151,7 +151,7 @@ static struct device_info boards[] = {
 		},
 
 		.first_sysupgrade_partition = "os-image",
-		.last_sysupgrade_partition = "file-system",
+		.last_sysupgrade_partition = "support-list",
 	},
 
 	/** Firmware layout for the CPE510/520 */
@@ -177,10 +177,10 @@ static struct device_info boards[] = {
 			{"default-mac", 0x30000, 0x00020},
 			{"product-info", 0x31100, 0x00100},
 			{"signature", 0x32000, 0x00400},
-			{"os-image", 0x40000, 0x170000},
-			{"soft-version", 0x1b0000, 0x00100},
-			{"support-list", 0x1b1000, 0x00400},
-			{"file-system", 0x1c0000, 0x600000},
+			{"os-image", 0x40000, 0x180000},
+			{"file-system", 0x1c0000, 0x5f0000},
+			{"soft-version", 0x7b0000, 0x00100},
+			{"support-list", 0x7b1000, 0x00400},
 			{"user-config", 0x7c0000, 0x10000},
 			{"default-config", 0x7d0000, 0x10000},
 			{"log", 0x7e0000, 0x10000},
@@ -189,7 +189,7 @@ static struct device_info boards[] = {
 		},
 
 		.first_sysupgrade_partition = "os-image",
-		.last_sysupgrade_partition = "file-system",
+		.last_sysupgrade_partition = "support-list",
 	},
 
 	{
@@ -209,10 +209,10 @@ static struct device_info boards[] = {
 			{"default-mac", 0x30000, 0x00020},
 			{"product-info", 0x31100, 0x00100},
 			{"signature", 0x32000, 0x00400},
-			{"os-image", 0x40000, 0x170000},
-			{"soft-version", 0x1b0000, 0x00100},
-			{"support-list", 0x1b1000, 0x00400},
-			{"file-system", 0x1c0000, 0x600000},
+			{"os-image", 0x40000, 0x180000},
+			{"file-system", 0x1c0000, 0x5f0000},
+			{"soft-version", 0x7b0000, 0x00100},
+			{"support-list", 0x7b1000, 0x00400},
 			{"user-config", 0x7c0000, 0x10000},
 			{"default-config", 0x7d0000, 0x10000},
 			{"log", 0x7e0000, 0x10000},
@@ -221,7 +221,7 @@ static struct device_info boards[] = {
 		},
 
 		.first_sysupgrade_partition = "os-image",
-		.last_sysupgrade_partition = "file-system",
+		.last_sysupgrade_partition = "support-list",
 	},
 
 	{
@@ -241,10 +241,10 @@ static struct device_info boards[] = {
 			{"default-mac", 0x30000, 0x00020},
 			{"product-info", 0x31100, 0x00100},
 			{"signature", 0x32000, 0x00400},
-			{"os-image", 0x40000, 0x170000},
-			{"soft-version", 0x1b0000, 0x00100},
-			{"support-list", 0x1b1000, 0x00400},
-			{"file-system", 0x1c0000, 0x600000},
+			{"os-image", 0x40000, 0x180000},
+			{"file-system", 0x1c0000, 0x5f0000},
+			{"soft-version", 0x7b0000, 0x00100},
+			{"support-list", 0x7b1000, 0x00400},
 			{"user-config", 0x7c0000, 0x10000},
 			{"default-config", 0x7d0000, 0x10000},
 			{"log", 0x7e0000, 0x10000},
@@ -253,7 +253,7 @@ static struct device_info boards[] = {
 		},
 
 		.first_sysupgrade_partition = "os-image",
-		.last_sysupgrade_partition = "file-system",
+		.last_sysupgrade_partition = "support-list",
 	},
 
 	/** Firmware layout for the C2600 */
